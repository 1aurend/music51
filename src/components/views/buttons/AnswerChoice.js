import React from 'react'
import { Button } from 'shards-react'
import styled from 'styled-components'
import {ButtonBorder} from '../layouts/PixelBorder'


//
// const StyledChoiceButton = styled(Button)`
  // min-height: 10vh;
  // min-width: 10vh;
  // margin-left: 2%;
  // margin-right: 2%;
  // margin-bottom: 2%;
  // font-size: 2rem;
  // text-align: center;
//   font-family: 'Overpass Mono', monospace;
//   font-weight: 600;
//   border-width: 4px;
//   border-radius: 0;
//   background-color: ${props => props.color};
//   &:hover {
//     background-color: ${props => props.color};
//   }
// `

const StyledChoiceButton = styled(Button)`
  display: block;
  cursor: pointer;
  border: none;
  padding: 0;
  background-color: ${props => props.color};
  &:hover {
    background-color: ${props => props.color};
  }
  min-height: 6vh;
  min-width: 6vh;
  font-size: 2rem;
  text-align: center;
`

const StyledKeystrokeSymbol = styled.p`
  font-size: 14px;
  color: #FF0000;
  margin-bottom: 0;
  margin-top: 2;
`

export default function AnswerChoice({ choice, keystroke, input, colors, onClick }) {
  const background = (() => {
    const thisInput = colors[colors.length-1]
    const greens = colors.filter(input => input.color === 'green').map(input => input.input)
    if (colors.length > 0 && thisInput.color === 'red' && thisInput.input === choice) {
      return '#B53F04'
    } else if (greens.includes(choice)) {
      return '#6AD97A'
    }
    return '#e5e6eb'
  })()

  function formatButtonText(choice) {
    if (choice.includes('^')) {
      return (
        <span>
          &nbsp;{choice.charAt(0)}
          <sup style={{position:'relative', left:'-15px', top:'-17px'}}>
            ^
          </sup>
        </span>
      )
    } else if (choice.includes('6') || choice.includes('4')) {
      return (
        <>
        {choice.slice(0,-2)}
          <span style={{postion: 'absolute'}}>
            <sup style={{display:'inline-block', position:'relative', left:'0px', top:'-17px'}}>
              {choice.charAt(choice.length-2)}
            </sup>
            <sub style={{position:'relative', left:'-14px', top:'6px'}}>
              {choice.charAt(choice.length-1)}
            </sub>
          </span>
        </>
      )
    }
    return choice
  }
  const formattedChoice = formatButtonText(choice)

  return (
    <ButtonBorder>
      <StyledChoiceButton
        theme='dark'
        color={background}
        onClick={onClick}
        >
          {formattedChoice}
          <StyledKeystrokeSymbol>
            [{keystroke}]
          </StyledKeystrokeSymbol>
      </StyledChoiceButton>
    </ButtonBorder>
  )
}
