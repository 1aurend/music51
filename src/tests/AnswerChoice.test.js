import React from 'react'
import AnswerChoice from '../components/views/buttons/AnswerChoice'
import renderer from 'react-test-renderer'

it('<AnswerChoice /> renders correctly', () => {
  const tree = renderer
    .create(<AnswerChoice
              onClick={() => {console.log('selected a choice')}}
              choice='B'
              keystroke='b'
              input={null}
              colors={[]}
              />)
    .toJSON();
  expect(tree).toMatchSnapshot()
})
