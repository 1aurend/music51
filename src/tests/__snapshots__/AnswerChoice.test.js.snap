// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnswerChoice /> renders correctly 1`] = `
<button
  className="sc-AykKC bOmntQ btn btn-light"
  color="#e5e6eb"
  onClick={[Function]}
  type="button"
>
  B
  <p
    className="sc-AykKD iJrcuN"
  >
    [
    b
    ]
  </p>
</button>
`;
